autoload/denops/_internal/server/chan.vim

```diff
@@ -170,6 +170,18 @@ function! s:reconnect(options) abort
         \))
   try
     call s:connect(s:addr, a:options)
+  catch /Could not find constraint\|Could not find version of/
```

Comment:
Use `denops#_internal#echo#warn()` instead debug.

Reply:
I have changed it to use warn.

Reply:
> This probably won't be caught.

I don't think so.  Because of the error message.

```
[denops] Server stopped (1). Restarting...
[denops] Failed to connect channel ``: Vim(let):connection failed: tcp address must be host:port
```

The error messages are only printed the function.

```vim
  catch
    call denops#_internal#echo#debug(printf(
          \ 'Failed to reconnect channel `%s` [%d/%d]: %s',
          \ s:addr,
          \ s:reconnect_count,
          \ a:options.reconnect_threshold,
          \ v:exception,
          \))
    call s:schedule_reconnect(a:options)
  endtry
```

Reply:
Oh, my bad.

`denops#_internal#server#chan#connect()` is correct instead.

Reply:
The error message breaks down as follows:

Outputs at `s:on_exit()` in *autoload/denops/_internal/server/proc.vim*:
```
[denops] Server stopped (1). Restarting...
```

Outputs at `denops#_internal#server#chan#connect()`:
```
[denops] Failed to connect channel ``: Vim(let):connection failed: tcp address must be host:port
```

Note that it says ```channel ``:```, which means `addr` is an empty string, so the local server will not start correctly. This is another bug caused by missing error handling.
If `addr` is empty, `DenopsSystemProcessListen:addr` event should not be fired, but this is a different scope so a separate PR would be better.

Outputs at `s:on_stderr()` in *autoload/denops/_internal/server/proc.vim*:
```
[denops] error: Could not find constraint in the list of versions: @std/async@^1.0.1
[denops]   Specifier: jsr:@std/async@^1.0.1/deadline
[denops]     at file:///Users/giginet/.local/share/nvim/lazy/denops.vim/denops/@denops-private/cli.ts:5:26
```

Note that this is not a part of the connect() error message.

Reply:
Hm...